sdk/samples/iot/sample_pnp_thermostat_component.c

```diff
@@ -114,40 +136,21 @@ static az_result invoke_getMaxMinReport(
   size_t len = strftime(end_time_buffer, sizeof(end_time_buffer), iso_spec_time_format, timeinfo);
   az_span end_time_span = az_span_create((uint8_t*)end_time_buffer, (int32_t)len);
 
-  az_json_writer json_builder;
-  AZ_RETURN_IF_FAILED(az_json_writer_init(&json_builder, response, NULL));
   AZ_RETURN_IF_FAILED(build_command_response_payload(
-      thermostat_component, &json_builder, start_time_span, end_time_span, out_response));
-
-  return AZ_OK;
-}
-
-static az_result build_telemetry_message(
-    sample_pnp_thermostat_component* thermostat_component,
-    az_span payload,
-    az_span* out_payload)
-{
-  az_json_writer json_builder;
-  AZ_RETURN_IF_FAILED(az_json_writer_init(&json_builder, payload, NULL));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_begin_object(&json_builder));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(&json_builder, telemetry_name));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_double(
-      &json_builder, thermostat_component->current_temperature, DOUBLE_DECIMAL_PLACE_DIGITS));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_end_object(&json_builder));
-  *out_payload = az_json_writer_get_bytes_used_in_destination(&json_builder);
+      thermostat_component, start_time_span, end_time_span, response, out_response));
 
   return AZ_OK;
 }
 
-static az_result append_double(az_json_writer* json_writer, void* value)
+static az_result append_double(az_json_writer* jw, void* value)
```

Comment:
Losing type-safety with such void* to (double*) casts worry me :)
Can we do anything about this?

Reply:
Not right now. The callback has to be generic to allow a user to append anything they want (we can't typedef the callback to have a `double` for value since that then disallows a string). This will likely go away since the official plug and play API's will not use a callback dynamic.